index.js

```diff
@@ -73,77 +81,65 @@ request.cookie = function (str) {
   return cookies.parse(str)
 }
 
-request.defaults = function (options, requester) {
+function wrap (method, options, requester) {
 
-  if (typeof options === 'function') {
-    requester = options
-    options = {}
-  }
+  return function (uri, opts, callback) {
+    var params = initParams(uri, opts, callback)
 
-  var self = this
-  var wrap = function (method) {
-    var headerlessOptions = function (options) {
-      options = extend({}, options)
-      delete options.headers
-      return options
+    var headerlessOptions = extend({}, options)
+    delete headerlessOptions.headers
+    params.options = extend(headerlessOptions, params.options)
+
+    if (typeof method === 'string') {
+      params.options.method = (method === 'del' ? 'DELETE' : method.toUpperCase())
+      method = request[method]
     }
 
-    var getHeaders = function (params, options) {
-      return constructObject()
-        .extend(options.headers)
-        .extend(params.options.headers)
-        .done()
+    if (options.headers) {
+      var headers = extend({}, options.headers)
+      params.options.headers = extend(headers, params.options.headers)
     }
 
-    return function (uri, opts, callback) {
-      var params = initParams(uri, opts, callback)
-      params.options = extend(headerlessOptions(options), params.options)
+    if (isFunction(requester)) {
+      method = requester
+    }
 
-      if (typeof method === 'string') {
-        params.options.method = (method === 'del' ? 'DELETE' : method.toUpperCase())
-        method = request[method]
-      }
+    return method(params.options, params.callback)
+  }
+}
 
-      if (params.options.method === 'HEAD' && paramsHaveRequestBody(params)) {
-        throw new Error('HTTP HEAD requests MUST NOT include a request body.')
-      }
+request.defaults = function (options, requester) {
+  var self = this
 
-      if (options.headers) {
-        params.options.headers = getHeaders(params, options)
-      }
+  if (typeof options === 'function') {
+    requester = options
+    options = {}
+  }
 
-      if (isFunction(requester)) {
-        method = requester
-      }
+  var defaults      = wrap(self, options, requester)
 
-      return method(params.options, params.callback)
-    }
-  }
+  var verbs = ['get', 'head', 'post', 'put', 'patch', 'del']
+  verbs.forEach(function(verb) {
+    defaults[verb]  = wrap(verb, options, requester)
```

Comment:
Yep I agree, crossed my mind, but I decided to keep it that way, though I haven't tested how the code will look like with it. So I'll give it a try.